cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -220,37 +226,27 @@ runQueryTip (AnyConsensusModeParams cModeParams) network mOutFile = do
             case slotToEpoch tipSlotNo eraHistory of
               Left e -> throwE (ShelleyQueryCmdPastHorizon e)
               Right (epochNo, _, _) -> do
-                let tipTimeResult = first toJsonPastHorizonException $
-                                fst <$> getProgress tipSlotNo eraHistory
-
-                case fmap getSystemStart mSystemStart of
-                  Just systemStart -> do
-                    nowSeconds <- toRelativeTime (SystemStart systemStart) <$> liftIO getCurrentTime
-
-                    let tolerance = RelativeTime (secondsToNominalDiffTime 600)
-                    let jsonSyncProgress = fmap (flip (percentage tolerance) nowSeconds) tipTimeResult
-
-                    let output = encodePretty $ O.QueryTipOutput
-                          { O.chainTip = tip
-                          , O.era = anyEra
-                          , O.epoch = epochNo
-                          , O.syncProgress = jsonSyncProgress
-                          }
-
-                    case mOutFile of
-                      Just (OutputFile fpath) -> liftIO $ LBS.writeFile fpath output
-                      Nothing                 -> liftIO $ LBS.putStrLn        output
-                  Nothing -> do
-                    let output = encodePretty $ O.QueryTipOutput
-                          { O.chainTip = tip
-                          , O.era = anyEra
-                          , O.epoch = epochNo
-                          , O.syncProgress = Left "No progress information available"
-                          }
-
-                    case mOutFile of
-                      Just (OutputFile fpath) -> liftIO $ LBS.writeFile fpath output
-                      Nothing                 -> liftIO $ LBS.putStrLn        output
+                syncProgressResult <- runExceptT $ do
+                  systemStart <- fmap getSystemStart mSystemStart & nothingThrowE ShelleyQueryCmdSystemStartUnavailable
+                  nowSeconds <- toRelativeTime (SystemStart systemStart) <$> liftIO getCurrentTime
+                  tipTimeResult <- getProgress tipSlotNo eraHistory & bimap ShelleyQueryCmdPastHorizon fst & except
+
+                  let tolerance = RelativeTime (secondsToNominalDiffTime 600)
```

Comment:
Out of curiosity, why 600?

Reply:
10 minutes.  Had a discussion with @dcoutts where he said a node can be up to a few minutes behind and still be "up to date".  I picked a nice round number that seemed to satisfy requirements.